libraries/core_libs/storage/src/storage.cpp

```diff
@@ -589,22 +584,12 @@ bool DbStorage::transactionFinalized(trx_hash_t const& hash) {
 
 std::vector<bool> DbStorage::transactionsInDb(std::vector<trx_hash_t> const& trx_hashes) {
   std::vector<bool> result(trx_hashes.size(), false);
-
-  DbStorage::MultiGetQuery db_query(shared_from_this(), trx_hashes.size());
-  db_query.append(DbStorage::Columns::transactions, trx_hashes);
-  auto db_trxs_statuses = db_query.execute();
-  for (size_t idx = 0; idx < db_trxs_statuses.size(); idx++) {
-    auto& trx_raw_status = db_trxs_statuses[idx];
-    result[idx] = !trx_raw_status.empty();
-  }
-
-  db_query.append(DbStorage::Columns::trx_period, trx_hashes);
-  db_trxs_statuses = db_query.execute();
-  for (size_t idx = 0; idx < db_trxs_statuses.size(); idx++) {
-    auto& trx_raw_status = db_trxs_statuses[idx];
-    result[idx] = result[idx] || (!trx_raw_status.empty());
+  for (size_t i = 0; i < trx_hashes.size(); ++i) {
+    const auto key = trx_hashes[i].asBytes();
+    if (exist(toSlice(key), Columns::transactions) || exist(toSlice(key), Columns::trx_period)) {
```

Comment:
isnt it more probable that tx is rather already in Columns::trx_period than  Columns::transactions ? In such case we should change the order in this if

Reply:
nope, we use this call when we insert new verified transaction so they are more likely be part of `Columns::transactions`